Sources/TestsCommon/CommonTestUtils.swift

```diff
@@ -122,6 +122,86 @@ open class MongoSwiftTestCase: XCTestCase {
     }
 }
 
+/// Enumerates the different topology configurations that are used throughout the tests
+public enum TestTopologyConfiguration: String, Decodable {
+    case sharded
+    case replicaSet
```

Comment:
From looking at the [evergreen failures](https://evergreen.mongodb.com/task_log_raw/mongo_swift_driver_tests_all__os_fully_featured~ubuntu_18.04_swift_version~5.1_ssl_auth~nossl_noauth_test_latest_replica_set_patch_2ce49dbfa192c24a9c02842ea0d8d7f50538a903_5efe2b4a3e8e861a667e11f1_20_07_02_18_45_31/0?type=T#L5611), it looks like the JSON tests actually specify this string as "replicaset" (no camelCase). To alleviate this, you can just specify the string value for this case (e.g. `case replicaSet = "replicaset"` ).